extensions/amp-analytics-resources/0.1/analytics-resources-impl.js

```diff
@@ -0,0 +1,192 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {getService} from '../../../src/service';
+import {parseUrl} from '../../../src/url';
+import {
+  getResourceTiming as rtcGetResourceTiming,
+} from 'resourcetiming-compression/src/resourcetiming-compression';
+
+/** @private Visible for testing. */
+export class AnalyticsResourcesService {
+  constructor(win) {
+    /** @private @const {!Window} */
+    this.win_ = win;
+  }
+
+  /**
+   * Returns rendered byte count of base page. This may not be what was sent on the network.
+   * @return {!Promise<string|undefined>}
```

Comment:
Here and below: is there any reason we shouldn't just return the actual types? E.g. `number`?